### PR TITLE
SC-738-v5-limits

### PR DESCRIPTION
**Pull request by potan**
Actions and data counts.

### CODE REVIEW

**Review by msmolyakov — APPROVED** [review]

File: node/src/main/scala/com/wavesplatform/api/http/UtilsApiRoute.scala [hunk 1]
```diff
@@ -317,6 +317,8 @@ object UtilsApiRoute {
               Set(),
               limitedExecution = false,
               remainingCalls = ContractLimits.MaxSyncDAppCalls(script.stdLibVersion),
+              avaliableActions = ContractLimits.MaxCallableActionsAmount(V5),
+              avaliableData = ContractLimits.MaxWriteSetSize(V5),
```
Comment (xrtm000) [inline node/src/main/scala/com/wavesplatform/api/http/UtilsApiRoute.scala]:
```suggestion
              avaliableData = ContractLimits.MaxWriteSetSize(script.stdLibVersion),
```

File: node/src/main/scala/com/wavesplatform/api/http/UtilsApiRoute.scala [hunk 2]
```diff
@@ -317,6 +317,8 @@ object UtilsApiRoute {
               Set(),
               limitedExecution = false,
               remainingCalls = ContractLimits.MaxSyncDAppCalls(script.stdLibVersion),
+              avaliableActions = ContractLimits.MaxCallableActionsAmount(V5),
```
Comment (xrtm000) [inline node/src/main/scala/com/wavesplatform/api/http/UtilsApiRoute.scala]:
```suggestion
              avaliableActions = ContractLimits.MaxCallableActionsAmount(script.stdLibVersion),
```

File: node/src/main/scala/com/wavesplatform/state/diffs/invoke/InvokeScriptDiff.scala [hunk 3]
```diff
@@ -188,18 +190,47 @@ object InvokeScriptDiff {
...
+                  val actionsCount = transfers.length
+                  if(actionsCount > avaliableActions) {
+                    val usedComplexity = remainingComplexity - unusedComplexity
+                    val error          = FailedTransactionError.dAppExecution("Actions count limit is exceeded", usedComplexity, log)
+                    traced(error.asLeft[(Diff, EVALUATED, Int, Int)])
+                  } else {
+                    doProcessActions(dataItems ::: transfers, unusedComplexity).map((_, unit, avaliableActions - actionsCount, avaliableData - dataCount))
+                  }
+                }
+              case ScriptResultV4(actions, unusedComplexity, ret) =>
```
Comment (xrtm000) [inline node/src/main/scala/com/wavesplatform/state/diffs/invoke/InvokeScriptDiff.scala:213]: Code below seems as duplication from `case ScriptResultV3` block. Please, extract to method and reuse if it possible

File: node/src/main/scala/com/wavesplatform/state/diffs/invoke/InvokeScriptTransactionDiff.scala [hunk 4]
```diff
@@ -181,8 +183,30 @@ object InvokeScriptTransactionDiff {
...
+                  TracedResult(Left(FailedTransactionError.dAppExecution("Stored data count limit is exceeded", fixedInvocationComplexity, log)))
+                } else {
+                  val actionsCount = transfers.length
+                  if(actionsCount > avaliableActions) {
+                    TracedResult(Left(FailedTransactionError.dAppExecution("Actions count limit is exceeded", fixedInvocationComplexity, log)))
+                  } else {
+                    doProcessActions(dataItems ::: transfers, unusedComplexity)
+                  }
+                }
+              case ScriptResultV4(actions, unusedComplexity, _) =>
```
Comment (xrtm000) [inline node/src/main/scala/com/wavesplatform/state/diffs/invoke/InvokeScriptTransactionDiff.scala]: Seems as duplication

File: node/src/test/scala/com/wavesplatform/state/diffs/ci/InvokeScriptV5LimitsTest.scala [hunk 5]
```diff
@@ -0,0 +1,794 @@
...
+//import com.wavesplatform.transaction.smart.script.ScriptCompiler
+//import com.wavesplatform.transaction.smart.script.trace.{AssetVerifierTrace, InvokeScriptTrace}
+import com.wavesplatform.transaction.smart.{InvokeScriptTransaction, SetScriptTransaction}
+//import com.wavesplatform.transaction.transfer.TransferTransaction
+import com.wavesplatform.transaction.{Asset, _}
+//import com.wavesplatform.utils._
+import com.wavesplatform.{NoShrink, TransactionGen}
+import org.scalacheck.Gen
+import org.scalamock.scalatest.MockFactory
+//import org.scalatest.exceptions.TestFailedException
```
Comment (xrtm000) [inline node/src/test/scala/com/wavesplatform/state/diffs/ci/InvokeScriptV5LimitsTest.scala]: Remove commented imports

**Review by xrtm000 — CHANGES_REQUESTED** [review]

**Review by xrtm000 — APPROVED** [review]

**Review by mrkraft — APPROVED** [review]

**Pull request merged**